### DISCUSSION TITLE
Place sports bet with GraphQL

### DISCUSSION THREAD

**Discussion by frafi1** [cloudbet/docs#28] [Q&A]
I got a "MALFORMED_REQUEST" trying to place a sports bet with the API.
What is wrong here? Could anyone tell me the proper formed request?

The query:
const input = {
   referenceId: "6177E56C-72AF-4B5F-8493-9AEBFAB89928",
   eventId: "12161278",
   price: "1.3",
   currency: "PLAY_EUR",
   marketUrl: "soccer.total_goals_period_second_half/under?total=2.25",
   stake: "0.1",
 };

 const query = `mutation PlaceBet($input: PlaceBetInput!) {
   placeBet(input: $input) {
     referenceId
     eventId
     marketUrl
     currency
     price
     stake
     betStatus
     betErrorCode
   }
 }`;
 
The response:
{
 "data":{
  "placeBet":{
   "referenceId":"6177E56C-72AF-4B5F-8493-9AEBFAB89928",
   "eventId":"12161278",
   "marketUrl":"soccer.total_goals_period_second_half/under?total=2.25",
   "currency":"PLAY_EUR",
   "price":"1.3",
   "stake":"0.1",
   "betStatus":"REJECTED",
   "betErrorCode":"MALFORMED_REQUEST"
  }
 }
}

**Comment by kgravenreuth** [comment]
Hi @frafi1 

Thanks for reporting this issue with our GraphQL Cloudbet API.

This seems to have been a bug on our side where we were mis-handling the `referenceId` passed here. Our engineering team has just pushed a fix. Could you please try again with a new betting request?

By the way, our GraphQL API blog post might give you more guidance on how to use our [GraphQL Betting API](https://www.cloudbet.com/en/blog/posts/sports-betting-graphql-api-with-javascript-node-js?utm_source=github.com&utm_medium=social&utm_campaign=axis) 

Please let us know if you see any other issues.

Regards,
Klaus Gravenreuth
Product & Engineering @ Cloudbet

**Reply by frafi1** [reply]
Hello Klaus,

now it works with the referenceId at lowercase.
The "MALFORMED_REQUEST" was a bit misleading, i searched only at the structure and could not find something wrong.

Franz
